src/components/admin/album/AlbumDetails.js

```diff
@@ -169,6 +170,32 @@ export default class AlbumDetails extends Component {
     }
   }
 
+  handleSetCoverPicClick(id, index){
+    var self = this
+    setCoverPhoto(id)
+    .then(function(response) {
+      self.handleSuccessResponse(response, index);
+    })
+    .catch(function(error) {
+      console.log(error.response);
+    });
+  }
+
+  handleSuccessResponse(response, index) {
+    var data = response.data
+    console.log(this.state.album)
+    if(response.status === 201){
+      const { cover_photo, photos } = this.state.album
+      photos.splice(index,1,cover_photo)
+      const setCoverPic = this.state.album
+      setCoverPic.cover_photo = data.data.photos;
+      const addCoverPhoto = Object.assign({}, this.state.album);
+      this.setState({album: addCoverPhoto});
```

Comment:
Please replace this code
```
const setCoverPic = this.state.album
setCoverPic.cover_photo = data.data.photos;
const addCoverPhoto = Object.assign({}, this.state.album);
this.setState({album: addCoverPhoto});
```

By
```
const newAlbum = Object.assign({}, this.state.album);
newAlbum.cover_photo = data.data.photo
this.setState({album: newAlbum});
```

And also change `set_cover_photo API` response, from `photos` to 'photo'